Make exhibits accessible only within a certain range.

thielep opened the pull request:
* Added an access radius in km to exhibits.
Such that exhibits are only shown if the users location is within the access radius of the exhibit.

*Added option to filter queries by location with given latitude and longitude coordinates.
Works for exhibits only right now, since only exhibits have an access radius.
If no coordinates are specified in a request, all exhibits are returned.

*Added the possibility to add latitude and longitude as exhibit query arguments.

*Added access radius to exhibit results.

Commit:
* Added an access radius in km to exhibits.
Such that exhibits are only shown if the users location is within the access radius of the exhibit.

*Added option to filter queries by location with given latitude and longitude coordinates.
Works for exhibits only right now, since only exhibits have an access radius.
If no coordinates are specified in a request, all exhibits are returned.

*Added the possibility to add latitude and longitude as exhibit query arguments.

*Added access radius to exhibit results.

SvenEV commented on HiP-DataStore/Controllers/QueryHelper.cs:
```diff
@@ -51,13 +51,56 @@ public static class QueryHelper
+        /// </summary>
+        public static IQueryable<T> FilterByLocation<T>(this IQueryable<T> query, float? latitude, float? longitude) where T : ContentBase
+        {
+            List<int> excludedIds = new List<int>();
```
You can shorten variable declarations using the [`var`-keyword](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/var), e.g.
```CSharp
var excludedIds = new List<int>()
```
As a rule of thumb and for consistency, please use `var` whenever possible.

SvenEV commented on HiP-DataStore/Controllers/QueryHelper.cs:
```diff
@@ -51,13 +51,56 @@ public static class QueryHelper
+        {
+            List<int> excludedIds = new List<int>();
+
+            if (typeof(Exhibit) == typeof(T))
```
Since the method is only applicable to exhibits (at the moment), consider making the method non-generic:
```CSharp
public static IQueryable<Exhibit> FilterByLocation(this IQueryable<Exhibit> query, ...)
```
Then, with some LINQ-magic, the code could be reduced to something like:
```CSharp
public static IQueryable<Exhibit> FilterByLocation(this IQueryable<Exhibit> query, float? latitude, float? longitude)
{
    if (latitude != null && longitude != null)
    {
        return query.ToList()
            .Where(o => GetDistanceFromLatLonInKm(o.Latitude, o.Longitude, latitude.Value, longitude.Value) <= o.AccessRadius)
            .AsQueryable();
    }

    return query;
}
```

SvenEV commented on HiP-DataStore/Controllers/QueryHelper.cs:
```diff
@@ -51,13 +51,56 @@ public static class QueryHelper
+        /// <summary>
+        /// Calculates the distance between two points of latitude and longitude in km.
+        /// </summary>
+        private static double GetDistanceFromLatLonInKm(float? lat1, float? lon1, float? lat2, float? lon2)
```
Are you sure you want the arguments to be of type `float?` (that is, [`Nullable<float>`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types))?. In your code you are ultimately casting them to `double`, which might cause an exception if one of the specified arguments is `null`.
Consider using just `float`, which will also make all the `double`-casts unnecessary.

*Pro tip*: Adding `using static System.Math;` to the beginning of the file lets you shorten all `Math.Bla(...)`-calls to just `Bla(...)` 😉

SvenEV reviewed (changes requested):
Some things can be simplified, but overall it looks fine! 

* ReSharper warnings need to be fixed
* Note to myself: For the existing exhibits, we have to write a migration which assigns them a default AccessRadius

Commit:
*Updated FilterByLocation method.
Type is now Exhibit and not generic, since it is only applicable to exhibits until now.
Also shortened it.

*Updated GetDistanceFromLatLonInKm method.
Arguments are no longer nullable.
Removed casts to double.

*Added Math library.

SvenEV committed:
Fixed ReSharper warnings

SvenEV reviewed (changes requested):
I forgot: We need a migration for existing exhibits

SvenEV committed:
Merge branch 'develop' into iss-hipcms-794-exhibit-location-access

SvenEV committed:
Added a migration that assigns a default access radius for "old" exhibits

SvenEV committed:
Merge branch 'iss-hipcms-794-exhibit-location-access' of https://github.com/HiP-App/HiP-DataStore into iss-hipcms-794-exhibit-location-access

SvenEV committed:
Merge branch 'develop' into iss-hipcms-794-exhibit-location-access

SvenEV committed:
Fixed ReSharper warning

SvenEV committed:
Merge branch 'iss-hipcms-794-exhibit-location-access' of https://github.com/HiP-App/HiP-DataStore into iss-hipcms-794-exhibit-location-access

SvenEV committed:
Fixed ReSharper warning
